modules/uipath/src/backend/index.ts

```diff
@@ -0,0 +1,105 @@
+import * as sdk from 'botpress/sdk'
+import jsonwebtoken from 'jsonwebtoken'
+
+export class ResponseError extends Error {
+  errorCode: string
+  statusCode: number
+
+  skipLogging = false
+
+  constructor(message: string, statusCode: number, errorCode: string) {
+    super(message)
+    Error.captureStackTrace(this, this.constructor)
+    this.statusCode = statusCode
+    this.errorCode = errorCode
+  }
+}
+
+class UnauthorizedError extends ResponseError {
+  type = 'UnauthorizedError'
+
+  constructor(message: string) {
+    super(`Unauthorized: ${message}`, 401, 'BP_0041')
+  }
+}
+
+const onServerStarted = async (bp: typeof sdk) => {
+  const authRouter = bp.http.createRouterForBot('uipath/auth', {
+    checkAuthentication: true
+  })
+
+  const rootRouter = bp.http.createRouterForBot('uipath', {
```

Comment:
What do you think of using "securedRouter" and "unsecuredRouter" instead ?

Reply:
Both are secured, they just don't use the same authentication strategy

Reply:
That's not true. rootRouter is not secured, while `/message` is secured because you added an auth middleware. If you forget to add the middleware, the route is not secured. The naming is to prevent that confusion.